csv_formatter.py

```diff
@@ -0,0 +1,309 @@
+import pandas as pd
+import pycountry
+import re
+
+
+def convert_upc_ean13(upc):
+    try:
+        upc = str(int(upc))
+    except ValueError:
+        return None
+    upc = "0" * (12 - len(upc)) + upc
+    ean13 = "0" + upc[:2] + "-" + upc[2:11] + "-" + upc[11]
+    return ean13
+
+
+def convert_country_alpha3(country_name):
+    spelling_changes = {"Vietnam": "Viet Nam", "Phillipines": "Philippines"}
+    if country_name in spelling_changes:
+        country_name = spelling_changes[country_name]
+    try:
+        return pycountry.countries.get(name=country_name).alpha_3
+    except Exception:
+        return None
+
+
+def check_prop_65(url_jpg, url_pdf):
+    if pd.isna(url_jpg) and pd.isna(url_pdf):
+        return False
+    else:
+        return True
+
+
+def check_attribute(data_cell, attribute):
+    if str(attribute).lower() in str(data_cell).lower():
+        return True
+    else:
+        return False
+
+
+def parse_dimension(dimensions, height, index):
+    if type(dimensions) == float:
+        return dimensions
+    else:
+        dimensions_array = re.split("x|X|\*", dimensions, maxsplit=2)
+        # After splitting, dimensions array may contain up to 3 values
+        # If there are three, 'height' is filtered out first
+        # Index is then used to determine which remaining value will serve as length or width
+        if len(dimensions_array) == 3:
+            height = float(height)
+            dimensions_array = list(map(lambda x: float(x), dimensions_array))
+            length_width = list(filter(lambda x: x != height, dimensions_array))
+            return float(length_width[index])
+        elif len(dimensions_array) == 2:
+            return float(dimensions_array[index])
+        else:
+            return float(dimensions_array[0])
+
+
+def format_currency(data_cell, currency_sign):
+    try:
+        return float(data_cell)
+    except ValueError:
+        data_cell = data_cell.replace(currency_sign, "")
+        data_cell = data_cell.replace(",", "")
+        return round(float(data_cell), 2)
+
+
+def convert_to_bool(data_cell):
+    data_cell = str(data_cell).lower()
+    return True if data_cell == "yes" else False
+
+
+with open("homework.csv", encoding="UTF-8") as input_file:
```

Comment:
Normal pattern for command line stuff in python

All of your actual code in a `main()`

```python
def main():
    ...
```

Down at the bottom, you want to invoke that function, only if you're directly calling the script

```python
if __name__ == "__main__":
    main()
```